libs/mocks/src/connectors.rs

```diff
@@ -0,0 +1,73 @@
+use cfg_traits::connectors::Codec;
+use codec::{Error, Input};
+
+#[derive(Debug, Eq, PartialEq)]
+pub enum MessageMock {
+	First,
+	Second,
+}
+
+impl MessageMock {
+	fn call_type(&self) -> u8 {
+		match self {
+			MessageMock::First => 0,
+			MessageMock::Second => 1,
+		}
+	}
+}
+
+impl Codec for MessageMock {
+	fn serialize(&self) -> Vec<u8> {
+		vec![self.call_type()]
+	}
+
+	fn deserialize<I: Input>(input: &mut I) -> Result<Self, Error> {
+		let call_type = input.read_byte()?;
+
+		match call_type {
+			0 => Ok(MessageMock::First),
+			1 => Ok(MessageMock::Second),
+			_ => Err("unsupported message".into()),
+		}
+	}
+}
```

Comment:
Maybe I'm going too far, but maybe a simple number can be used as a message to avoid this type?

An later use something like:
```rust
domain_router.send(sender, 42).unwrap()
```

Reply:
Mmm... that's weird 🤔 . I'm pretty sure `Codec` and `Serialize` is implemented for basic types. Could it be because the compiler doesn't know exactly what that `42` is? Sometimes in these cases, you need to help the compiler with something like `42u32`.

Reply:
You are probably referring to a different `Codec` trait, I'm referring to the one that we have in connectors - https://github.com/centrifuge/centrifuge-chain/blob/add-pallet-connectors-gateway/libs/traits/src/connectors.rs#L19

Reply:
Ouch, you're right! Never mind then

Reply:
Although... would it make sense to add the basic types to the `Codec` trait along with the implementation (as it happens with the substrate `Coded` and `Serialize` traits)? @NunoAlexandre @wischli